platform/darwin/test/MGLStyleLayerTests.m

```diff
@@ -19,4 +25,65 @@ - (void)setUp {
 #endif
 }
 
+- (void)testPropertyName:(NSString *)name isBoolean:(BOOL)isBoolean {
+    NS_MUTABLE_ARRAY_OF(NSString *) *components = [name componentsSeparatedByString:@"-"].mutableCopy;
+    if (isBoolean) {
+        if ([components.firstObject isEqualToString:@"is"]) {
+            [components removeObjectAtIndex:0];
+            if (![components.lastObject.lexicalClasses containsObject:NSLinguisticTagAdjective]) {
+                XCTAssertTrue([components.lastObject.lexicalClasses containsObject:NSLinguisticTagVerb],
+                              @"Boolean getter %@ that starts with “is” should contain an adjective, past participle, or verb.", name);
+                XCTAssertNotEqualObjects(components.lastObject.lemma, components.lastObject,
+                                         @"Boolean getter %@ should not have infinitive, imperative, or present tense verb.", name);
+            }
+        } else {
+            if ([components.firstObject isEqualToString:[self class].layerType]
+                || [components.firstObject isEqualToString:@"icon"] || [components.firstObject isEqualToString:@"text"]) {
+                [components removeObjectAtIndex:0];
+            }
+#if TEST_STRICT_NAMING_CONVENTIONS
+            XCTAssertTrue([components.firstObject.lexicalClasses containsObject:NSLinguisticTagVerb],
+                          @"Boolean getter %@ that doesn’t start with “is” should contain a verb.", name);
+            XCTAssertNotEqualObjects(components.firstObject.lemma, components.lastObject);
+#endif
+        }
+    } else {
+        XCTAssertFalse([components.firstObject isEqualToString:@"is"]);
+#if TEST_STRICT_NAMING_CONVENTIONS
+        XCTAssertTrue([components.lastObject.lexicalClasses containsObject:NSLinguisticTagNoun],
+                      @"Non-Boolean getter %@ should contain a noun.", name);
+#endif
+    }
+}
+
+@end
+
+@implementation NSString (MGLStyleLayerTestAdditions)
+
+- (NS_ARRAY_OF(NSString *) *)lexicalClasses {
+    NSOrthography *orthography = [NSOrthography orthographyWithDominantScript:@"Latn"
```

Comment:
Note that this method doesn't get used for much by default. I put most calls to this method behind a compile-time flag that's off by default, because NSLinguisticTagging is actually pretty bad at tagging parts of speech for individual words. We could do better by moving these tests into JavaScript as part of the codegen script, but unfortunately the most popular npm tagging modules either require calls to a Web API or are licensed under the GPL.